### ISSUE TITLE
Make OpenEth OpenSource

### ISSUE THREAD

**Issue by recmo** [dex-ethics/open-eth#6]
"can we make a public OpenEth Repo by any chance?"

**Comment by recmo** [comment]
We need to be explicit about the licenses. There are two licenses we need to specify. One for the source code (MIT, GPL or somesort) and one for the content on the website (a flavour of CC).

The most open would be MIT License and CC BY 4.0. Note that these allow commercial use of the content by anyone.

@NellInc What do you think?

**Comment by NellInc** [comment]
I agree that MIT or BSD license seems ideal for code and CC for content.

CC BY 4.0 seems like a good blend, assuming that BY attribution can be collective, rather than needing to be explicit for each piece of content, or at least would be trivial enough to add in that function through an API. 

It also ensures accountability and traceability, though the open version would not come with any warranties for end users whatsoever.

MIT and BSD are very similar, except that BSD offers a clause whereby one's use of code cannot be an implied endorsement by the entity who originally produced it. This is probably not likely to be a concern for us.

Now MIT and GPL are different since GPL offers more guarantees that the code cannot be incorporated within proprietary software. Open Source is about true freedom, and that includes the freedom to use code within commercial projects. This link has quite an eloquent discussion: https://ttmm.io/tech/why-the-mit-license/

In short, I concur that MIT + CC BY 4.0 seems appropriate.

**Comment by recmo** [comment]
> CC BY 4.0 seems like a good blend, assuming that BY attribution can be collective, rather than needing to be explicit for each piece of content, or at least would be trivial enough to add in that function through an API.

Do we want the CC BY 4.0 to apply to user supplied content as well? If so, there is a legal technicality that users need to sign over the copyright of submitted content to us. Or at the very least they need to provide it under the same CC BY 4.0 to us. This later option sounds appealing, but it has the disadvantage that the original author needs to be involved in changing the license later (for whatever reason). Some open source projects that did not do this have spend lots of time tracking down the original authors to have them sign legal mundanities.

We could include a message in the bottom stating that "By using this site you agree to us sharing your content under the license we use for Open Eth content, currently CC BY 4.0."

**Comment by recmo** [comment]
Wikipedia is a good role-model for this. In the bottom line there is:

>  By using this site, you agree to the Terms of Use and Privacy Policy.

Which links to

https://wikimediafoundation.org/wiki/Terms_of_Use#7._Licensing_of_Content

**Comment by NellInc** [comment]
Agreed, Wikipedia seems like the best model here.
